heartbeat/gcp-vpc-move-route.in

```diff
@@ -0,0 +1,442 @@
+      'ip', 'iproute', 'project', 'route_name', 'vpc_network', \
+      'vpc_network_url', 'zone'
+
+
+def wait_for_operation(ctx, response):
+  """Blocks until operation completes.
+  Code from GitHub's GoogleCloudPlatform/python-docs-samples
+
+  Args:
+    response: dict, a request's response
+  """
+  def _OperationGetter(response):
+    operation = response[u'name']
+    if response.get(u'zone'):
+      return ctx.conn.zoneOperations().get(
+          project=ctx.project, zone=ctx.zone, operation=operation)
+    else:
+      return ctx.conn.globalOperations().get(
+          project=ctx.project, operation=operation)
+
+  while True:
+    result = _OperationGetter(response).execute()
+
+    if result['status'] == 'DONE':
+      if 'error' in result:
+        raise Exception(result['error'])
+      return result
+
+    time.sleep(1)
+
+
+def get_metadata(metadata_key, params=None, timeout=None):
+  """Performs a GET request with the metadata headers.
+
+  Args:
+    metadata_key: string, the metadata to perform a GET request on.
+    params: dictionary, the query parameters in the GET request.
+    timeout: int, timeout in seconds for metadata requests.
+
+  Returns:
+    HTTP response from the GET request.
+
+  Raises:
+    urlerror.HTTPError: raises when the GET request fails.
+  """
+  timeout = timeout or 60
+  metadata_url = os.path.join(METADATA_SERVER, metadata_key)
+  params = urlparse.urlencode(params or {})
+  url = '%s?%s' % (metadata_url, params)
+  request = urlrequest.Request(url, headers=METADATA_HEADERS)
+  request_opener = urlrequest.build_opener(urlrequest.ProxyHandler({}))
+  return request_opener.open(request, timeout=timeout * 1.1).read()
+
+
+def validate(ctx):
+  if os.geteuid() != 0:
+    logging.error('You must run this agent as root')
+    sys.exit(OCF_ERR_PERM)
+
+  try:
+    ctx.conn = googleapiclient.discovery.build('compute', 'v1')
+  except Exception as e:
+    logging.error('Couldn\'t connect with google api: ' + str(e))
+    sys.exit(OCF_ERR_CONFIGURED)
+
+  ctx.ip = os.environ.get('OCF_RESKEY_ip')
+  if not ctx.ip:
+    logging.error('Missing ip parameter')
+    sys.exit(OCF_ERR_CONFIGURED)
+
+  try:
+    ctx.instance = get_metadata('instance/name')
+    ctx.zone = get_metadata('instance/zone').split('/')[-1]
+    ctx.project = get_metadata('project/project-id')
+  except Exception as e:
+    logging.error(
+        'Instance information not found. Is this a GCE instance ?: %s', str(e))
+    sys.exit(OCF_ERR_CONFIGURED)
+
+  ctx.instance_url = '%s/projects/%s/zones/%s/instances/%s' % (
+      GCP_API_URL_PREFIX, ctx.project, ctx.zone, ctx.instance)
+  ctx.vpc_network = os.environ.get('OCF_RESKEY_vpc_network', 'default')
+  ctx.vpc_network_url = '%s/projects/%s/global/networks/%s' % (
+      GCP_API_URL_PREFIX, ctx.project, ctx.vpc_network)
+  ctx.interface = os.environ.get('OCF_RESKEY_interface', 'eth0')
+  ctx.route_name = os.environ.get(
+      'OCF_RESKEY_route_name', 'ra-%s' % os.environ['__SCRIPT_NAME'])
```

Comment:
should I change here to "os.path.basename(sys.argv[0])" too?

Reply:
You might as well for consistency.

Reply:
done